CircleciScripts/generate_documentation.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
+# Creates Swift and Objective-C documentation by module, packages all modules
+# into a ZIP file, and leaves the result in the ${DOCS_TMP} directory, which is
+# safe to leave while checking out the gh-pages directory
+
+set -x
+
+SDK_VERSION="2.12.3"
+
+GITHUB_DOC_ROOT=https://aws-amplify.github.io
+GITHUB_SOURCE_ROOT=https://github.com/aws-amplify/aws-sdk-ios
+
+if [[ -z $SOURCE_ROOT ]] ; then
+  SOURCE_ROOT=$( pwd )
+fi
+
+DOCS_TMP=${SOURCE_ROOT}/docs_tmp
+
+# remove everything generated by this script
+function cleanup {
+  rm -rf ${DOCS_TMP}
+}
+
+cleanup
+
+if [[ $1 == "clean" ]] ; then
+	exit 0
+fi
+
+cd "$SOURCE_ROOT"
+
+mkdir -p ${DOCS_TMP}
+
+test_device_id=$( bash ${SOURCE_ROOT}/CircleciScripts/get_circleci_test_device_id.sh )
+exitValue=$?
+if [[ $exitValue -ne  0 ]] || [[ -z $test_device_id ]] ; then
+  echo "Error getting test device ID"
+  exit $exitValue
```

Comment:
nit: probably better to return `1` in the event that get_circleci_test_device_id.sh returns 0, but an error?

Reply:
D'oh! And here I thought I was being soooo clever. :(